contracts/chain-adapters/SP1_Adapter.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./interfaces/AdapterInterface.sol";
+
+contract DataStore {
+    error NotHubPool();
+    mapping(address => bytes) public dataForTarget;
+    address public immutable hubPool;
+
+    modifier onlyHubPool() {
+        if (msg.sender != hubPool) {
+            revert NotHubPool();
+        }
+        _;
+    }
+
+    constructor(address _hubPool) {
+        hubPool = _hubPool;
+    }
+
+    function storeData(address target, bytes calldata data) external onlyHubPool {
+        dataForTarget[target] = data;
+    }
+}
+
+/**
+ * @notice Stores data that can be relayed to L2 SpokePool using SP1 + Helios light clients.
+ */
+contract SP1_Adapter is AdapterInterface {
+    DataStore public immutable DATA_STORE;
+
+    constructor(DataStore _store) {
+        DATA_STORE = _store;
+    }
+
+    /**
+     * @notice Saves root bundle data in a simple storage contract that can be proven and relayed to L2.
+     * @param target Contract on the destination that will receive the message.
+     * @param message Data to send to target.
+     */
+    function relayMessage(address target, bytes calldata message) external payable override {
+        DATA_STORE.storeData(target, abi.encode(address(this), target, message));
+        emit MessageRelayed(target, message);
+    }
+
+    /**
+     * @notice No-op relay tokens method.
+     */
+    function relayTokens(
+        address,
+        address,
+        uint256,
+        address
+    ) external payable override {
+        // This method is intentionally left as a no-op.
+        // If the adapter is intended to be able to relay tokens, this method should be overridden.
```

Comment:
+1 great point on overriding

We may want to include CCTP, OFTs, xERC20, etc (all the token standards we support) in some derived class in case we want to enable some token rebalancing routes to these chains where canonical bridges exist.

Reply:
Yeah, I think it would be good to have the capability to use any token standard we already support by default. Would that work?

Reply:
i don't see how that can be done cleanly for any token standard. We can have this adapter extend the OFTAdapter and/or the HypeXERC20Adapter but each new standard we'll have to update this contract. WDYT?

Reply:
We can add in a follow up PR, this PR is too large

Reply:
and the HypERC20 code isn't complete yet

Reply:
The main consideration is that this PR then depends on the OFT/HypXERC20 PR's being merged and therefore reviewed along with this PR, right?

Reply:
To remove dependency on OFT/HypXERC20 unaudited PR's, we've decided to extend CCTP adapter with this PR and then follow up in a separate PR that adds OFT/HypXERC20 support to this PR once merged into master